torch/csrc/jit/script/init.cpp

```diff
@@ -77,12 +77,17 @@ struct VISIBILITY_HIDDEN PythonValue : public SugaredValue {
       throw ErrorReport(loc) << "keyword arguments in Python calls aren't supported";
     Graph& g = *m.graph();
 
-    // this python object might be a @trace or @script stand-alone function
+    // this python object might be a @trace or @script function/module
     // if so, inline the graph rather than calling the python
-    if(py::isinstance<GraphExecutor>(self)) {
-      GraphExecutor& ge = py::cast<GraphExecutor&>(self);
-      ensureSizeMatches(loc, ge.graph()->inputs().size(), inputs.size(), "arguments");
-      return packOutputs(*m.graph(),inlineCallTo(*m.graph(), *ge.graph(), inputs));
+
+    if(py::isinstance<Module>(self)) {
```

Comment:
This code is unsafe: We keep ModuleValues separate from PythonValues because the only legit module calls are to submodules which are obtained via attribute lookup from self. Otherwise you can call a non-submodule whose parameters are not owned by the surrounding module. Those parameters can then go out of scope and crash the program. 

There needs to be a check that the "module" being accessed this way has no parameters before the call to emit_call_to is valid.